chore: silence unhelpful forge lints

TomAFrench opened the pull request:
Foundry has been screaming at me whenever I bootstrap the l1-contracts with what looks like a lot of low-value lints.

I've shied away from making any code changes as I don't think it would be particularly helpful so this PR mostly boils down to silencing lints coming from dependencies and disabling lints which just are just code style-guide items.

One semi-meaningful thing which I've disabled is a lint calling out an opportunities to optimize keccak hashing so I've opened an issue to track this here: https://github.com/AztecProtocol/aztec-packages/issues/16808

`forge lint` now outputs the below. This looks like we can add `/// forge-lint: disable-next-item(erc20-unchecked-transfer)` to these cases as we control the token being used in these situations -  but I'll leave that up to the tmnts.

```
warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
  --> /mnt/user-data/tom/aztec-packages/l1-contracts/src/core/libraries/rollup/RewardLib.sol:90:5
   |
90 |     rollupStore.config.feeAsset.transfer(_sequencer, amount);
   |     --------------------------------------------------------
   |
   = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
   --> /mnt/user-data/tom/aztec-packages/l1-contracts/src/core/libraries/rollup/StakingLib.sol:184:5
    |
184 |     store.stakingAsset.transfer(exit.recipientOrWithdrawer, exit.amount);
    |     --------------------------------------------------------------------
    |
    = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
   --> /mnt/user-data/tom/aztec-packages/l1-contracts/src/core/libraries/rollup/StakingLib.sol:285:5
    |
285 |     store.stakingAsset.transferFrom(msg.sender, address(this), amount);
    |     ------------------------------------------------------------------
    |
    = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
   --> /mnt/user-data/tom/aztec-packages/l1-contracts/src/core/libraries/rollup/StakingLib.sol:357:9
    |
357 |         store.stakingAsset.transfer(args.withdrawer, amount);
    |         ----------------------------------------------------
    |
    = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
   --> /mnt/user-data/tom/aztec-packages/l1-contracts/src/core/libraries/rollup/RewardLib.sol:122:5
    |
122 |     rollupStore.config.feeAsset.transfer(_prover, accumulatedRewards);
    |     -----------------------------------------------------------------
    |
    = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
   --> /mnt/user-data/tom/aztec-packages/l1-contracts/src/governance/GSE.sol:339:5
    |
339 |     ASSET.transferFrom(msg.sender, address(this), ACTIVATION_THRESHOLD);
    |     -------------------------------------------------------------------
    |
    = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
   --> /mnt/user-data/tom/aztec-packages/l1-contracts/src/governance/GSE.sol:472:5
    |
472 |     ASSET.transferFrom(msg.sender, address(this), amount);
    |     -----------------------------------------------------
    |
    = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
   --> /mnt/user-data/tom/aztec-packages/l1-contracts/src/core/libraries/rollup/RewardLib.sol:218:9
    |
218 |         rollupStore.config.feeAsset.transfer(BURN_ADDRESS, t.totalBurn);
    |         ---------------------------------------------------------------
    |
    = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
   --> /mnt/user-data/tom/aztec-packages/l1-contracts/test/portals/TokenPortal.sol:149:5
    |
149 |     underlying.transfer(_recipient, _amount);
    |     ----------------------------------------
    |
    = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

```

LHerskind commented:
Thanks Tom. Will take a look. A lot of it seems to be linting the tests as well 👍

TomAFrench committed:
chore: silence unhelpful forge lints

Foundry has been screaming at me whenever I bootstrap the l1-contracts with what looks like a lot of low-value lints.

I've shied away from making any code changes as I don't think it would be particularly helpful so this PR mostly boils down to silencing lints coming from dependencies and disabling lints which just are just code style-guide items.

One semi-meaningful thing which I've disabled is a lint calling out an opportunities to optimize keccak hashing so I've opened an issue to track this here: https://github.com/AztecProtocol/aztec-packages/issues/16808

`forge lint` now outputs the below. This looks like we can add `/// forge-lint: disable-next-item(erc20-unchecked-transfer)` to these cases as we control the token being used in these situations -  but I'll leave that up to the tmnts.

```
warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
  --> /mnt/user-data/tom/aztec-packages/l1-contracts/src/core/libraries/rollup/RewardLib.sol:90:5
   |
90 |     rollupStore.config.feeAsset.transfer(_sequencer, amount);
   |     --------------------------------------------------------
   |
   = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
   --> /mnt/user-data/tom/aztec-packages/l1-contracts/src/core/libraries/rollup/StakingLib.sol:184:5
    |
184 |     store.stakingAsset.transfer(exit.recipientOrWithdrawer, exit.amount);
    |     --------------------------------------------------------------------
    |
    = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
   --> /mnt/user-data/tom/aztec-packages/l1-contracts/src/core/libraries/rollup/StakingLib.sol:285:5
    |
285 |     store.stakingAsset.transferFrom(msg.sender, address(this), amount);
    |     ------------------------------------------------------------------
    |
    = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
   --> /mnt/user-data/tom/aztec-packages/l1-contracts/src/core/libraries/rollup/StakingLib.sol:357:9
    |
357 |         store.stakingAsset.transfer(args.withdrawer, amount);
    |         ----------------------------------------------------
    |
    = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
   --> /mnt/user-data/tom/aztec-packages/l1-contracts/src/core/libraries/rollup/RewardLib.sol:122:5
    |
122 |     rollupStore.config.feeAsset.transfer(_prover, accumulatedRewards);
    |     -----------------------------------------------------------------
    |
    = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
   --> /mnt/user-data/tom/aztec-packages/l1-contracts/src/governance/GSE.sol:339:5
    |
339 |     ASSET.transferFrom(msg.sender, address(this), ACTIVATION_THRESHOLD);
    |     -------------------------------------------------------------------
    |
    = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
   --> /mnt/user-data/tom/aztec-packages/l1-contracts/src/governance/GSE.sol:472:5
    |
472 |     ASSET.transferFrom(msg.sender, address(this), amount);
    |     -----------------------------------------------------
    |
    = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
   --> /mnt/user-data/tom/aztec-packages/l1-contracts/src/core/libraries/rollup/RewardLib.sol:218:9
    |
218 |         rollupStore.config.feeAsset.transfer(BURN_ADDRESS, t.totalBurn);
    |         ---------------------------------------------------------------
    |
    = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

warning[erc20-unchecked-transfer]: ERC20 'transfer' and 'transferFrom' calls should check the return value
   --> /mnt/user-data/tom/aztec-packages/l1-contracts/test/portals/TokenPortal.sol:149:5
    |
149 |     underlying.transfer(_recipient, _amount);
    |     ----------------------------------------
    |
    = help: https://book.getfoundry.sh/reference/forge/forge-lint#erc20-unchecked-transfer

```
